views/helpers/annotations.js

```diff
@@ -0,0 +1,157 @@
+  const [yMin, yMax] = extent.getExtents(hexagons, ({ y }) => y);
+
+  annotations.forEach(a => {
+    let hexagon = hexagons.find(h => h.text[0] === a.region);
+
+    if(hexagon) {
+      let horizontalIncrement = hexagon.width/4;
+      let verticalIncrement = hexagon.height/4;
+
+      if (a.position === "top" || a.position === "left") {
+        // If contentWidth (cssModifier) is narrow, all annotations on the left will be drawn on the top
+        a.coordinates = getTopCoordinates(hexagon.x, hexagon.y, yMin, horizontalIncrement, verticalIncrement, annotationStartPosition);
+        
+        if (cssModifier !== "narrow" && a.position === "left") {
+          a.coordinates = getLeftCoordinates(hexagon.x, hexagon.y, xMin, verticalIncrement, annotationStartPosition);
+        }
+      } else {
+        // If contentWidth (cssModifier) is narrow, all annotations on the right will be drawn on the bottom
+        a.coordinates = getBottomCoordinates(hexagon.x, hexagon.y, yMax, hexagon.height, horizontalIncrement, verticalIncrement, annotationStartPosition);
+        
+        if (cssModifier !== "narrow" && a.position === "right") {
+          a.coordinates = getRightCoordinates(hexagon.x, hexagon.y, xMax, hexagon.width, verticalIncrement, annotationStartPosition);
+        }
+      }
+    }
+  });
+  return annotations;
+}
+
+/**
+ * Links the annotations to the features of the geo map and sets all the coordinates needed for drawing the annotations correctly
+ */
+function setCoordinatesForGeoMap(annotations, geoParameters, entityType, annotationStartPosition, cssModifier) {
+  let path = geoParameters.path;
+  let features = geoParameters.features.features;
+  let yMax = geoParameters.bounds[1][1];
+  let xMax = geoParameters.bounds[1][0];
+
+  annotations.forEach(a => {
+    let feature = features.find(f => f.properties[entityType] === a.region);
+
+    if (feature) {
+      let centroid = path.centroid(feature);
+
+      if (a.position === "top" || a.position === "left") {
+        // If contentWidth (cssModifier) is narrow, all annotations on the left will be drawn on the top
+        a.coordinates = getTopCoordinates(centroid[0], centroid[1], 0, 0, 0, annotationStartPosition);
+        
+        if (cssModifier !== "narrow" && a.position === "left") {
+          a.coordinates = getLeftCoordinates(centroid[0], centroid[1], 0, 0, annotationStartPosition);
+        }
+      } else {
+        // If contentWidth (cssModifier) is narrow, all annotations on the right will be drawn on the bottom
+        a.coordinates = getBottomCoordinates(centroid[0], centroid[1], yMax, 0, 0, 0, annotationStartPosition);
+        
+        if (cssModifier !== "narrow" && a.position === "right") {
+          a.coordinates = getRightCoordinates(centroid[0], centroid[1], xMax, 0, 0, annotationStartPosition);
+        }
+      }
+    }
+  });
+  return annotations;
+}
+
+/**
+ * Helper function for setCoordinatesForHexMap() and setCoordinatesForGeoMap()
+ */
```

Comment:
Since this is a helper-file, I don't think it's necessary to describe this here. Can be deleted imo.